pyomo/contrib/interior_point/interface.py

```diff
@@ -0,0 +1,624 @@
+        pass
+
+    @abstractmethod
+    def evaluate_eq_constraints(self):
+        pass
+
+    @abstractmethod
+    def evaluate_ineq_constraints(self):
+        pass
+
+    @abstractmethod
+    def evaluate_jacobian_eq(self):
+        pass
+
+    @abstractmethod
+    def evaluate_jacobian_ineq(self):
+        pass
+
+    @abstractmethod
+    def init_slacks(self):
+        pass
+
+    @abstractmethod
+    def init_duals_primals_lb(self):
+        pass
+
+    @abstractmethod
+    def init_duals_primals_ub(self):
+        pass
+
+    @abstractmethod
+    def init_duals_slacks_lb(self):
+        pass
+
+    @abstractmethod
+    def init_duals_slacks_ub(self):
+        pass
+
+    @abstractmethod
+    def set_slacks(self, slacks):
+        pass
+
+    @abstractmethod
+    def set_duals_primals_lb(self, duals):
+        pass
+
+    @abstractmethod
+    def set_duals_primals_ub(self, duals):
+        pass
+
+    @abstractmethod
+    def set_duals_slacks_lb(self, duals):
+        pass
+
+    @abstractmethod
+    def set_duals_slacks_ub(self, duals):
+        pass
+
+    @abstractmethod
+    def get_slacks(self):
+        pass
+
+    @abstractmethod
+    def get_duals_primals_lb(self):
+        pass
+
+    @abstractmethod
+    def get_duals_primals_ub(self):
+        pass
+
+    @abstractmethod
+    def get_duals_slacks_lb(self):
+        pass
+
+    @abstractmethod
+    def get_duals_slacks_ub(self):
+        pass
+
+    @abstractmethod
+    def set_barrier_parameter(self, barrier):
+        pass
+
+    @abstractmethod
+    def evaluate_primal_dual_kkt_matrix(self, timer=None):
+        pass
+
+    @abstractmethod
+    def evaluate_primal_dual_kkt_rhs(self, timer=None):
+        pass
+
+    @abstractmethod
+    def set_primal_dual_kkt_solution(self, sol):
+        pass
+
+    @abstractmethod
+    def get_delta_primals(self):
+        pass
+
+    @abstractmethod
+    def get_delta_slacks(self):
+        pass
+
+    @abstractmethod
+    def get_delta_duals_eq(self):
+        pass
+
+    @abstractmethod
+    def get_delta_duals_ineq(self):
+        pass
+
+    @abstractmethod
+    def get_delta_duals_primals_lb(self):
+        pass
+
+    @abstractmethod
+    def get_delta_duals_primals_ub(self):
+        pass
+
+    @abstractmethod
+    def get_delta_duals_slacks_lb(self):
+        pass
+
+    @abstractmethod
+    def get_delta_duals_slacks_ub(self):
+        pass
+
+    def regularize_equality_gradient(self, kkt, coef, copy_kkt=True):
+        raise RuntimeError(
+            'Equality gradient regularization is necessary but no '
+            'function has been implemented for doing so.')
+
+    def regularize_hessian(self, kkt, coef, copy_kkt=True):
+        raise RuntimeError(
+            'Hessian of Lagrangian regularization is necessary but no '
+            'function has been implemented for doing so.')
+
+
+class InteriorPointInterface(BaseInteriorPointInterface):
+    def __init__(self, pyomo_model):
+        if type(pyomo_model) is str:
```

Comment:
Should this take in a PyNumero NLP instead of a pyomo model or ampl model? I don't see the code making use of anything out of the higher level models.

Reply:
Good idea.

Reply:
There are two methods that make use of features of a Pynumero `PyomoNLP`: `_get_full_duals_primals_bounds` and `load_primals_into_pyomo_model`. Should these methods be moved elsewhere? Should we require a `PyomoNLP`? I'm not sure what the best thing to do here is.

Reply:
I was wrong. There are several methods at the bottom of the class that implement methods from the `PyomoNLP`. Maybe we should have a class that inherits from this class for Pyomo models? `InteriorPointInterface` and `PyomoInteriorPointInterface`? Then we can move these methods to the derived class?